Added ShippingLines, TaxLines, TaxLine, Gateway to Order

spprichard opened the pull request:
- Made structure for ShippingLines
- Made structure for TaxLines
- Made structure for TaxLine
- Added Gateway to Order structure


Need ability to get shipping information for a Order. This would allow for information that is available in Order/X.json, to be in this project. 

Use Case could include calculating the shipping cost for a specific order.

Commit:
Added ShippingLines, TaxLines, TaxLine

- Made stucture for ShippingLines
- Made structure for TaxLines
- Made structure for TaxLine

jerairrest commented on order.go:
```diff
@@ -69,7 +69,7 @@ type Order struct {
 	Note                  string           `json:"note"`
 	Test                  bool             `json:"test"`
-	BrowserIp             string           `json:"browser_ip"`
+	BrowserIP             string           `json:"browser_ip"`
```
This would be a breaking change

jerairrest replied:
We maybe want to just leave this value

dlebech replied:
It's my bad that I didn't initially name this according to convention, but now that it's there, I would prefer not making a breaking change for a stylistic detail.

spprichard replied:
No problem. I'll switch it back

jerairrest commented on order.go:
```diff
@@ -137,6 +137,25 @@ type OrdersResource struct {
+	Code                string           `json:"code"`
+	Source              string           `json:"source"`
+	Phone               string           `json:"phone"`
+	RequestedFufillment string           `json:"requested_fulfillment_service_id"`
```
I would say be verbose

dlebech replied:
I agree, verbose is better. There is also a tiny typo, a missing `l` in in `Fufillment`. I suggest:
`RequestedFulfillmentServiceID`

spprichard replied:
That makes sense. I've changed it

jmnsf reviewed (commented):
I would also agree that we should probably have a 1-1 mapping between JSON keys and struct fields as it's less error-prone for users.

dlebech commented on order.go:
```diff
@@ -137,6 +137,25 @@ type OrdersResource struct {
+	Phone               string           `json:"phone"`
+	RequestedFufillment string           `json:"requested_fulfillment_service_id"`
+	DeliveryCategory    string           `json:"delivery_category"`
+	CarrierID           string           `json:"carrier_identifier"`
```
I would also suggest following Shopify's naming here, like `CarrierIdentifier`

spprichard replied:
Will do. Changed

dlebech reviewed (commented):
Thank you for the PR! It looks like you didn't add the shipping line struct as a slice to the order struct. Was this intentional?

Commit:
Changes after review

- Chagned BrowserIP back to BrowserIp becuase breaking changes are bad
- Renamed ShippingLines structure to match Shopift convention
- Added ShippingLines structure to Order structure
- Refactored name RequestedFufillment to RequestedFufillmentServiceID

dlebech reviewed (approved):
Thanks @spprichard . Looks good to me

Commit:
Added Gateway to order struct

- The gateway string is a part of orders/X.json, but not a part of the
Order Structure

dlebech commented:
Hehe, did you sneak in an extra field after my review @spprichard ? :smile: Are you ok for me to merge this now?

dlebech commented:
By the way, the `gateway` field is deprecated so you probably shouldn't use it. In fact, I would suggest not adding it, since it can be removed by Shopify any time, and probably already doesn't have the data you would expect from it.

dlebech commented on order.go:
```diff
@@ -69,12 +69,14 @@ type Order struct {
 	OrderNumber           int              `json:"order_number"`
 	Note                  string           `json:"note"`
 	Test                  bool             `json:"test"`
+	Gateway               string           `json:"gateway"`
```
As I mentioned elsewhere, this field is deprecated since 2014, so I don't think we should add it so we can avoid someone becoming reliant on it.

Commit:
Removing Depricated Feild

- Removed Gateway

spprichard commented:
@dlebech I removed Gateway. Should be good to merge now if you think all is good. Thanks!

dlebech commented:
Merged now. Thanks again @spprichard
